lib/rules/unit-disallowed-list/__tests__/index.js

```diff
@@ -388,6 +388,67 @@ testRule({
 	],
 });
 
+testRule({
+	ruleName,
+
+	config: [
+		['px'],
+		{
+			ignoreFunctions: ['calc', /^translate/],
+		},
+	],
+
+	accept: [
+		{
+			code: 'a { margin: calc(50% - 100px) }',
+		},
+		{
+			code: 'a { translateX: calc(50% - 100px) }',
+		},
+		{
+			code: 'a { translateY: calc(50% - 100px) }',
+		},
+		{
+			code: 'a { translateZ: calc(50% - 100px) }',
+		},
```

Comment:
[suggest] A case for `/^translate/` seems necessary, and duplicate cases seems unnecessary:

```suggestion
		{
			code: 'a { transform: translateX(100px) }',
		},
```